### DISCUSSION TITLE
Adding FTXUI into app with existing logs

### DISCUSSION THREAD

**Discussion by AlexKordic** [ArthurSonzogni/FTXUI#936] [Q&A] [answered]
Hi,

I have experimented with screen rendering after app logs using my crude methods. 

What is the best way to add app logging to `ScreenInteractive`? 

Following is my goal:

Logs get printed first, then `n` blank lines follow and then screen is placed. When mouse is disabled user can scroll up and read all logs while UI is updated at the bottom.

I am picturing moving cursor up by the `(screen-y-size + n)`, then printing log lines, then redrawing screen.

For printing batch of log lines I would like to wait 0.2sec after log to redraw screen delaying UI until log is active. In this case logs would overwrite leftover screen, this looks ok/readable. Additionally `n` blank lines serve as blank background for low frequency logs.

Can I use `ScreenInteractive::TrackMouse(false)` to disable mouse from within key event handler? Would I need to exit `ScreenInteractive::Loop()` for changes to take effect?

There is [isatty()](https://www.man7.org/linux/man-pages/man3/isatty.3.html) to check for terminal presence and if absent I would disable the UI.

Thanks,
Alex

**Comment by AlexKordic** [comment]
Seams like `ScreenInteractive::set_cursor_position` contains string to position cursor to top row of a screen.

Can't figure out where terminal gets erased.

**Comment by ArthurSonzogni** [comment, marked as answer]
Hello!

Maybe you could try wrapping the function logging output to the console with:
`Closure ScreenInteractive::WithRestoredIO(Closure)`?


When calling the decorated function, the screen would "exit" before executing the inner function, and being "restored" after.

**Reply by AlexKordic** [reply]
Pure awesomeness!
